// @framer/jurrehoutkamp.catalina-dynamic/code/CatalinaDynamic.tsx
import * as React from "react"
import { PropertyControls, ControlType, Frame } from "framer"

interface Props {
    opacity: number
    image: string
    width: number
    height: number
}

export class CatalinaDynamic extends React.Component<Props> {
    // Set default properties
    static defaultProps = {
        opacity: 0,
        width: 960,
        height: 540,
    }

    // Items shown in property panel
    static propertyControls: PropertyControls = {
        opacity: {
            type: ControlType.Number,
            title: "TimeShift",
            defaultValue: 0,
            min: 0,
            max: 1,
            unit: "",
            step: 0.01,
        },
    }

    render() {
        const { width, height } = this.props
        return (
            <Frame
                style={{
                    width: width,
                    height: height,
                }}
            >
                <Frame
                    style={{
                        width: width,
                        height: height,
                        backgroundImage: `url(https://946663360620-us-east-1-static-files.s3.amazonaws.com/assets/platforms/catalina_dynamic/night.jpg)`,
                        backgroundSize: "cover",
                        backgroundPosition: "center",
                        display: "flex",
                        justifyContent: "center",
                        alignItems: "center",
                        opacity: this.props.opacity,
                        position: "absolute",
                        top: "0",
                        left: "0",
                        zIndex: 99,
                        transition: "opacity 0.5s ease-out",
                    }}
                />

                <Frame
                    style={{
                        width: width,
                        height: height,
                        backgroundImage: `url(https://946663360620-us-east-1-static-files.s3.amazonaws.com/assets/platforms/catalina_dynamic/day.jpg)`,
                        backgroundSize: "cover",
                        backgroundPosition: "center",
                        display: "block",
                        justifyContent: "center",
                        alignItems: "center",
                        position: "absolute",
                        top: "0",
                        left: "0",
                    }}
                />
            </Frame>
        )
    }
}
