// @framer/monika-m.embedded/code/App.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"

// CSS Styles
const style: React.CSSProperties = {
    height: "100%",
    width: "100%",
    display: "flex",
    overflow: "hidden",
    borderRadius: "8px",
    background: "rgba(136, 85, 255, 0.1)",
    alignItems: "center",
    justifyContent: "center",
    fontSize: 16,
    fontWeight: 600,
    color: "#BB88FF",
    flexDirection: "column",
    textAlign: "center",
}

// Intstructions

function ComponentInstructions() {
    return (
        <div style={style}>
            <p> 🖼 </p>
            <p style={{ margin: 0, marginBottom: 5 }}>
                {" "}
                Add a URL in the properties panel
            </p>
            <p
                style={{
                    margin: 0,
                    fontWeight: 500,
                    opacity: 0.5,
                    fontSize: 14,
                    width: 440,
                }}
            >
                Make sure the given web address starts with <b>http://</b> or{" "}
                <b>https://</b>.
            </p>
        </div>
    )
}

// Condition

function isUrlInvalid(text: string): boolean {
    const startsWithHttp =
        text.startsWith("http://") || text.startsWith("https://")

    if (text === undefined || text === null || text === "" || !startsWithHttp) {
        return true
    } else {
        return false
    }
}

type Props = {
    text: string
    radius: number
}

// Component

export function Embedded(props: Props) {
    if (isUrlInvalid(props.text)) {
        return <ComponentInstructions />
    }

    return (
        <Frame style={{ ...style, borderRadius: props.radius }}>
            <iframe
                src={props.text}
                frameBorder={0}
                style={{ width: "100%", height: "100%" }}
            ></iframe>
        </Frame>
    )
}

// Default Props

Embedded.defaultProps = {
    text: "",
    radius: 8,
    width: 800,
    height: 600,
}

// Property Controls

addPropertyControls(Embedded, {
    text: {
        type: ControlType.String,
        title: "URL",
        defaultValue: Embedded.defaultProps.text,
    },
    radius: {
        type: ControlType.Number,
        title: "Radius",
        defaultValue: Embedded.defaultProps.radius,
    },
})
